packages/lodestar-validator/src/services/attestation.ts

```diff
@@ -314,7 +309,7 @@ export class AttestationService {
       computeEpochAtSlot(this.config, aggregate.data.slot)
     );
     const signingRoot = computeSigningRoot(this.config, this.config.types.AggregateAndProof, aggregateAndProof, domain);
-    return this.secretKeys[aggregatorIndex].sign(signingRoot).toBytes();
+    return this.validatorClients[aggregatorIndex].secretKey.sign(signingRoot).toBytes();
```

Comment:
I wonder if we should add utility method to fetch validator by pubkey from our array of validators. This is quite confusing because in eth2 context aggregatorIndex is index of validator in validator registry not in array of our validators

Reply:
or better yet instead of array have hexpubkey => {validator, secretKey} map so it's faster in case we are having multiple validators

Reply:
I'm not sure I follow.  Why/how would we want to fetch by pubkey if we are given a `number` index?  We could also maybe just call the variable `validatorStore` instead of `validators` if that is the confusing part.

Reply:
Index in our validators array which we call throughout block and attestation service is not the same index as in eth2 spec is confusing part.
Example:
We are running 6 validators in our validator client. There is total of 10k validators in beacon chain.
If we wanna take some action with validator in position 2 in our client. We will store "2" as value in variable called validatorIndex or aggregatorIndex but in beacon context that's compleatly different validator. This is especially confusing where duty.validatorIndex is not the index of validator in our array.

So we should have map pubkey => {secretKey, validator} so our code is more clear and we can find validator faster since validator is identified with pubkey in most of api responses

Reply:
just pushed a change for this